app/models/assignment_repo/creator.rb

```diff
@@ -132,6 +134,26 @@ def delete_github_repository(github_repo_id)
       true
     end
 
+    # Public: Clone the GitHub template repository for the AssignmentRepo.
+    #
+    # Returns an Integer ID or raises a Result::Error
+    # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
+    def create_github_repository_from_template!
+      client = assignment.creator.github_client
+      repository_name = generate_github_repository_name
+      options = {
+        name: repository_name,
+        owner: organization.github_organization.login,
+        private: assignment.private?,
+        description: "#{repository_name} created by GitHub Classroom",
+        accept: "application/vnd.github.baptiste-preview"
+      }
+
+      client.post("https://api.github.com/repositories/#{assignment.starter_code_repo_id}/generate", options)
```

Comment:
Could `https://api.github.com` be pulled out to a constant, like `GITHUB_API_HOST`?

Reply:
Good idea. I'll also move the preview accept header into a constant